### PR TITLE
Allow user to choose Function runtime

### PR DESCRIPTION
**Pull request by aerfio**
**Description**

Changes proposed in this pull request:

- UI part of https://github.com/kyma-project/kyma/issues/9055

**Related issue(s)**

https://github.com/kyma-project/kyma/issues/9055

### CODE REVIEW

File: core-ui/src/components/Lambdas/LambdasList/LambdasList.js [hunk 1]
```diff
@@ -41,8 +41,8 @@ function LambdaStatusBadge({ status }) {
   );
 }
 
-const headerRenderer = () => ['Name', 'Labels', 'Status'];
-const textSearchProperties = ['name', 'status.phase'];
+const headerRenderer = () => ['Name', 'Runtime', 'Labels', 'Status'];
+const textSearchProperties = ['name', 'runtime', 'status.phase'];
```
Comment (aerfio) [inline core-ui/src/components/Lambdas/LambdasList/LambdasList.js:45]: question for reviewer: is it possible to put some mapping function here?

File: core-ui/src/components/Lambdas/LambdaDetails/LambdaDetailsHeader/LambdaDetailsHeader.js [hunk 2]
```diff
@@ -43,6 +44,10 @@ export default function LambdaDetailsHeader({ lambda }) {
       <div className="fd-panel-grid fd-panel-grid--3col lambda-details-header__content">
         <div className="fd-has-grid-column-span-2">
           <LambdaLabels lambda={lambda} />
+          <div className="fd-has-margin-bottom-l">
+            <span className="fd-has-color-text-4 ">{'Runtime: '}</span>
```
Comment (magicmatatjahu) [inline core-ui/src/components/Lambdas/LambdaDetails/LambdaDetailsHeader/LambdaDetailsHeader.js:48]: `Runtime` to const :P

File: core-ui/src/components/Lambdas/LambdaDetails/Tabs/Code/CodeAndDependencies/CodeAndDependencies.js [hunk 3]
```diff
@@ -58,9 +63,16 @@ export default function CodeAndDependencies({ lambda }) {
     }
 
     const deps = (dependencies || '').trim();
-    if (!(deps.startsWith('{') && deps.endsWith('}'))) {
-      setDisabledCause(DISABLED_CAUSES.INVALID_DEPS);
-      return;
+
+    switch (lambda.runtime) {
```
Comment (magicmatatjahu) [inline core-ui/src/components/Lambdas/LambdaDetails/Tabs/Code/CodeAndDependencies/CodeAndDependencies.js:67]: Maybe make from this logic separate function in `helpers/runtime`? Also please create the `helpers/runtime` folder.

File: core-ui/src/components/Lambdas/LambdaDetails/LambdaDetailsHeader/LambdaDetailsHeader.js (hunk 2 above)
Comment (magicmatatjahu) [reply core-ui/src/components/Lambdas/LambdaDetails/LambdaDetailsHeader/LambdaDetailsHeader.js:48]: It should also be in `helpers/runtime/prettyRuntime` file :)

File: core-ui/src/components/Lambdas/LambdasList/LambdasList.js (hunk 1 above)
Comment (magicmatatjahu) [reply core-ui/src/components/Lambdas/LambdasList/LambdasList.js:45]: Mapping in which form?

File: core-ui/src/components/Lambdas/LambdasList/RuntimeDropdown.js [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+      <FormLabel htmlFor="runtime-dropdon">
+        {LAMBDAS_LIST.CREATE_MODAL.INPUTS.RUNTIME.NAME}
+      </FormLabel>
+      <FormSelect
+        ref={_ref}
+        id="runtime-dropdon"
+        role="select"
+        defaultValue={defaultRuntime}
+      >
+        {Object.values(functionAvailableLanguages).map(lang => (
```
Comment (magicmatatjahu) [inline core-ui/src/components/Lambdas/LambdasList/RuntimeDropdown.js:25]: `Object.entries(functionAvailableLanguages).map(([runtime, lang]) => {...})`

File: core-ui/src/components/Lambdas/config.js [hunk 5]
```diff
@@ -7,17 +7,28 @@ export const WEBHOOK_VALIDATION = {
   MIN_REQUEST_MEMORY: 'WEBHOOK_VALIDATION_MIN_REQUEST_MEMORY',
```
Comment (magicmatatjahu) [inline core-ui/src/components/Lambdas/config.js:7]: If you change something in this file, please also update changed values in this file https://github.com/kyma-project/kyma/blob/master/resources/console/charts/web/templates/configmap.yaml#L31 and https://github.com/kyma-project/kyma/blob/master/resources/serverless/charts/webhook/templates/configmap.yaml

File: core-ui/src/components/Lambdas/gql/hooks/mutations/useCreateLambda.js [hunk 6]
```diff
@@ -69,15 +72,11 @@ export const useCreateLambda = ({ redirect = true }) => {
...
-    lambdaName: name,
-  });
-
+export function prepareCreateLambdaInput(name, runtime = 'nodejs12') {
   return {
     labels: {},
-    source: CONFIG.defaultLambdaCode,
-    dependencies,
+    source: CONFIG.defaultLambdaCodeAndDeps[runtime].code,
+    dependencies: getDefaultDependencies(name, runtime),
```
Comment (magicmatatjahu) [inline core-ui/src/components/Lambdas/gql/hooks/mutations/useCreateLambda.js:79]: `CONFIG.defaultLambdaCodeAndDeps[runtime].deps` ? :D Is helper function needed?

**Review by magicmatatjahu — COMMENTED** [review]
Great job! I only have a problem with helpers for runtime in `misc` folder. Please create `helpers/runtime` and put inside the dir whole logic related to it.

File: core-ui/src/components/Lambdas/gql/hooks/mutations/useCreateLambda.js (hunk 6 above)
Comment (aerfio) [reply core-ui/src/components/Lambdas/gql/hooks/mutations/useCreateLambda.js:79]: yeah, I don't want to inline that logic

File: core-ui/src/components/Lambdas/constants.js [hunk 7]
```diff
@@ -327,7 +337,7 @@ export const CODE_AND_DEPENDENCIES_PANEL = {
     TEXT: 'Save',
     POPUP_MESSAGE: {
       EMPTY_SOURCE: 'Source cannot be empty.',
-      INVALID_DEPS: "Dependencies should start with '{' and end with '}'.",
+      INVALID_DEPS: 'Dependencies should be a valid JSON',
```
Comment (magicmatatjahu) [inline core-ui/src/components/Lambdas/constants.js:340]: JSON or plain text :P

**Review by magicmatatjahu — APPROVED** [review]

**Review by parostatkiem — APPROVED** [review]

**Pull request merged**